src/backend/cpu/kernel/mean.hpp

```diff
@@ -108,9 +109,10 @@ struct mean_dim<Ti, Tw, To, 0> {
         To* out            = output.get();
 
         dim_t istride = istrides[dim];
-        MeanOp<Ti, To, Tw> Op(0, 0);
-        for (dim_t i = 0; i < idims[dim]; i++) {
-            Op(in[inOffset + i * istride], 1);
+        dim_t end     = inOffset + idims[dim] * istride;
+        MeanOp<compute_t<Ti>, compute_t<To>, compute_t<Tw>> Op(0, 0);
```

Comment:
Shouldn't this start with the first element since you are starting with non-zero offset in the for loop below ?

Reply:
The offset determines the start of the axis where you want to perform the mean. You still want the mean to start anew from that point.